src/journal_entry.rs

```diff
@@ -51,22 +65,19 @@ impl JournalEntry {
 	}
 
 	pub fn get_source_wallclock_time(&self) -> Option<JournalEntryTimestamp> {
-		self.fields
-			.get("_SOURCE_REALTIME_TIMESTAMP")
+		self.get_field_string_lossy("_SOURCE_REALTIME_TIMESTAMP")
 			.and_then(|v| v.parse::<i64>().ok())
 			.map(|v| JournalEntryTimestamp { timestamp_us: v })
```

Comment:
```suggestion
		self.get_field_string("_SOURCE_REALTIME_TIMESTAMP").ok()
		    .flatten()
			.and_then(|v| v.parse::<i64>().ok())
			.map(|v| JournalEntryTimestamp { timestamp_us: v })
```

Minor performance note, if a string isn't valid utf8 it isn't going to be parsable as in integer so no need to allocate the string for the lossy version. (and same with the other time functions below)